### ISSUE TITLE
[require-description] checkConstructors not working when contexts sets to [MethodDefinition]

### ISSUE THREAD

**Issue by OpportunityLiu** [gajus/eslint-plugin-jsdoc#534]
For example:

``` js
/**
 *
 */
class Foo {
    /**
     *
     */
    constructor() {}

    /**
     *
     */
    bar() {}
}

```
I need to add [ClassDeclaration] to context for the doc of class, but with `contexts: [ClassDeclaration]`, both constructor & bar have no `Missing JSDoc block description.` warning. When I add [MethodDefinition] to `contexts`, or set `contexts: [any]`, both constructor & bar have `Missing JSDoc block description.` warning, even if `checkConstructors: false` is set.

**Comment by brettz9** [comment]
The `checkConstructors` option does not exist on `require-jsdoc`.

You can get the behavior, however, as follows (and I've added this as a test case / docs, for future reference) (the contexts are esquery expressions, and you can consult the esquery docs and the likes of https://astexplorer.net/ to discover AST details for building expressions for your desired syntax):

```js
        {
          contexts: [
            'MethodDefinition[key.name!="constructor"]',
          ],
          require: {
            ClassDeclaration: true,
          },
        }
```

Closing as that should address, but feel free to comment further.

**Comment by OpportunityLiu** [comment]
I means `require-description`, not `require-jsdoc`

https://github.com/gajus/eslint-plugin-jsdoc#eslint-plugin-jsdoc-rules-require-description

The checkConstructors option exists, otherwise eslint will failed to parse `.eslintrc` before it can actually analysis files.
